### DISCUSSION TITLE
Choose which branch is used in a manual build

### DISCUSSION THREAD

**Discussion by polyGeek** [codemagic-ci-cd/codemagic-docs#1112] [Q&A] [answered]
I built my Flutter app for Android and when I downloaded it from the store the version appeared to be the first commit I ever made to GitHub. It's not remotely the current version of the app.

Is there a way for me to specify which branch is used when using Build Manually? 

Note: I would like to build the most recent push to the "main" branch.

**Comment by himesh-cm** [comment, marked as answer]
Hi @polyGeek,
When you trigger a new build from the dashboard, you should see an option to select the branch and workflow.
Please check the attachment below.

<img width="576" alt="branch" src="https://user-images.githubusercontent.com/106069365/198949262-da292eba-41de-475e-94a8-4d3eb88be398.png">

Let us know if you still have any questions.

**Comment by polyGeek** [comment]
Well duh! That should have been obvious. Sorry, I thought it would be a setting that remained relatively static. This is much better. Thanks.

**Comment by kentnek** [comment]
Hi, is it possible to select a specific commit to build?
